flixel/input/actions/FlxActionManager.hx

```diff
@@ -0,0 +1,735 @@
+ * - UPDATE all your action sets at once
+ * - ENFORCE the "only one action set active per device at a time" rule
+ * 
+ * Once you've set up FlxActionManager, you can let flixel handle it globally
+ * with: FlxG.inputs.add(myFlxActionManager);
+ * 
+ * If you don't add it globally, you will have to call update() on it yourself.
+ * 
+ * If you are using the steamwrap library, FlxActionManager gains the ability
+ * to automatically create action sets from a steamwrap object derived from the
+ * master vdf game actions file that Steam makes you set up for native
+ * controller support. You must then ACTIVATE one of those action sets for any
+ * connected steam controllers, which will automatically attach the proper
+ * steam action inputs to the actions in the set. You can also add as many 
+ * regular FlxActionInputs as you like to any actions in the sets.
+ * 
+ * NOTE:
+ * If you are using a Steam Controller, you MUST use FlxActionManager in order
+ * to properly process the Steam Controller API via FlxActions. The only other
+ * alternative is to directly call the steamwrap functions directly.
+ * 
+ */
+ 
+class FlxActionManager implements IFlxInputManager implements IFlxDestroyable
+{
+	private var sets:Array<FlxActionSet>;
+	private var register:ActionSetRegister;
+	
+	/**
+	 * The number of registered action sets
+	 */
+	public var numSets(get, null):Int;
+	
+	public function new() 
+	{
+		sets = [];
+		register = new ActionSetRegister();
+	}
+	
+	/**
+	 * Activate an action set for a particular device
+	 * @param	ActionSet	The integer ID for the Action Set you want to activate
+	 * @param	Device		The device type (Mouse, Keyboard, Gamepad, SteamController, etc)
+	 * @param	DeviceID	FlxGamepad ID or a Steam Controller Handle (ignored for Mouse/Keyboard)
+	 */
+	public function activateSet(ActionSet:Int, Device:FlxInputDevice, ?DeviceID:Int = FlxInputDeviceID.ALL)
+	{
+		register.activate(ActionSet, Device, DeviceID);
+		onChange();
+	}
+	
+	/**
+	 * Add an action to a particular action set
+	 * @param	Action		The FlxActionDigital you want to add
+	 * @param	ActionSet	The index of the FlxActionSet you want to add
+	 * @return	whether it was successfully added
+	 */
+	public function addDigitalAction(Action:FlxActionDigital, ActionSet:Int):Bool
+	{
+		return addAction(Action, ActionSet, FlxInputType.DIGITAL);
+	}
+	
+	/**
+	 * Add an action to a particular action set
+	 * @param	Action		The FlxActionAnalog you want to add
+	 * @param	ActionSet	The index of the FlxActionSet you want to add
+	 * @return	whether it was successfully added
+	 */
+	public function addAnalogAction(Action:FlxActionAnalog, ActionSet:Int):Bool
+	{
+		return addAction(Action, ActionSet, FlxInputType.ANALOG);
+	}
+	
+	/**
+	 * Add a FlxActionSet to the manager
+	 * @param	set	The FlxActionSet you want to add
+	 * @return	the index of the FlxActionSet
+	 */
+	public function addSet(set:FlxActionSet):Int
+	{
+		for (s in sets)
```

Comment:
Should actually use `contains()` instead of `indexOf()` from `FlxArrayUtil` via static extension.